src/jit/lowerxarch.cpp

```diff
@@ -3320,6 +3344,11 @@ void Lowering::TreeNodeInfoInitCmp(GenTreePtr tree)
         {
             MakeSrcContained(tree, op1);
         }
+        else if (op1->IsCnsIntOrI())
+        {
+            // Codegen doesn't support swapping op1 and op2 to generate cmp R/M, imm.
+            SetRegOptional(op2);
```

Comment:
Done.

---
In reply to: [79527039](https://github.com/dotnet/coreclr/pull/7252#discussion_r79527039) [](ancestors = 79527039)